Wrong variable checked for encrypted pool flag in TskAutoDb::addUnallocFsSpaceToDb

When compiling with C++20, the following warning is emitted:
```
tsk/auto/auto_db.cpp: In member function 'TSK_RETVAL_ENUM TskAutoDb::addUnallocFsSpaceToDb(size_t&)':
tsk/auto/auto_db.cpp:1401:57: error: bitwise operation between different enumeration types 'TSK_VS_PART_FLAG_ENUM' and 'TSK_POOL_VOLUME_FLAGS' is deprecated [-Werror=deprecated-enum-enum-conversion]
 1401 |                                 if (curVsPartInfo.flags & TSK_POOL_VOLUME_FLAG_ENCRYPTED) {
      |                                     ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The problem here is that `curVsPartInfo.flags` and `TSK_POOL_VOLUME_FLAG_ENCRYPTED` are not from the same enum; it makes no sense to check for `TSK_POOL_VOLUME_FLAG_ENCRYPTED` in `curVsPartInfo.flags` because the latter has nothing to do with pools at all.

--- a/tsk/auto/auto_db.cpp
+++ b/tsk/auto/auto_db.cpp
@@ -1398,7 +1398,7 @@ TSK_RETVAL_ENUM TskAutoDb::addUnallocFsSpaceToDb(size_t & numFs) {
 
                             }
                             else {
-                                if (curVsPartInfo.flags & TSK_POOL_VOLUME_FLAG_ENCRYPTED) {
+                                if (pool->vol_list->flags & TSK_POOL_VOLUME_FLAG_ENCRYPTED) {
                                     tsk_error_reset();
                                     tsk_error_set_errno(TSK_ERR_FS_ENCRYPTED);
                                     tsk_error_set_errstr(

